### ISSUE TITLE
Connection placeholder content written to user settings file

### ISSUE THREAD

**Issue by af4jm** [microsoft/vscode-mssql#17065]
- MSSQL Extension Version: 1.10.1
- VSCode Version: 1.60.2
- OS Version: Win10

Steps to Reproduce: install extension, restart VSCode, see nonsense added to user settings

```json
    "mssql.connections": [
        {
            "server": "{{put-server-name-here}}",
            "database": "{{put-database-name-here}}",
            "user": "{{put-username-here}}",
            "password": "{{put-password-here}}"
        }
    ],
```

Expected behavior: setting should be there if and only if it contains actual connection data, the template makes sense in the default settings pane, not live user settings

**Comment by justin-dx** [comment]
I've just had the same thing happen after adding a new server. I got my password wrong on the first attempt and after getting it correct I end up with two connections.
![vscode mssql](https://user-images.githubusercontent.com/39646902/140836845-86ae3523-b64f-4500-9d54-aee27c98e169.JPG)

**Comment by kburtram** [comment]
@justin-dx we've observed this before and will fix in the next release.  In our repo the placeholder went away when vscode was restarted.  Please let us know if that's not what you also observed in your environment.

**Comment by kburtram** [comment]
This should be fixed in the 1.11.1 release.  Thanks!
